pilot/pkg/model/config.go

```diff
@@ -587,6 +587,32 @@ func ResolveShortnameToFQDN(host string, meta ConfigMeta) Hostname {
 	return Hostname(out)
 }
 
+// resolveGatewayName uses metadata information to resolve a reference
+// to shortname of the gateway to FQDN
+func resolveGatewayName(gwname string, meta ConfigMeta) string {
+	out := gwname
+
+	// New way of binding to a gateway in remote namespace
+	// is ns/name. Old way is either FQDN or short name
+	if !strings.Contains(gwname, "/") {
+		if !strings.Contains(gwname, ".") {
+			// we have a short name. Resolve to a gateway in same namespace
+			out = fmt.Sprintf("%s/%s", meta.Namespace, gwname)
```

Comment:
This does beg the question, why not allow short name since it's just a shorthand notation for `./name` and it's even backwards compatible?

Reply:
It is allowed. I should remove the deprecation warning